packages/components/src/components/combobox/utils.ts

```diff
@@ -49,3 +51,46 @@ export const filterOnlyOptgroups = (items: HTMLElement[]) => items.filter(isOptg
  * @returns The normalized string
  */
 export const normalizeString = (str: string) => str.normalize('NFD').replace(/[\u0300-\u036f]/g, '').toLowerCase();
+
+/**
+ * Creates a `SynOption` element from a given `TemplateResult`.
+ * We assume that the given `TemplateResult` is a <syn-option>.
+ *
+ * @param template - The `TemplateResult` to be rendered into a `SynOption`.
+ * @returns The created SynOption element
+ */
+export const createElementFromTemplateResult = (template: TemplateResult) => {
+  const container = document.createElement('div');
+  render(template, container);
+  return container.firstElementChild as HTMLElement;
+};
+
+/**
+ * This function takes an input of type `TemplateResult`, `string`, or `HTMLElement`
+ * and creates an SynOption from it.
+ *
+ * @param option - The option which can be of type `TemplateResult`, `string` or `HTMLElement`.
+ * @returns The created SynOption or undefined if the option was not a valid type.
+ */
+export const createOptionFromDifferentTypes = (option: TemplateResult | string | HTMLElement) => {
+  if (!option) return undefined;
+
+  const getOptionOrUndefined = (element: HTMLElement) => (element.tagName.toLocaleLowerCase() === 'syn-option' ? element as SynOption : undefined);
```

Comment:
This function will be created one time in memory for each call of `createOptionFromDifferentTypes`. As this is just a simple test, I would go with either:

a) Placing it above the `createOptionFromDifferentTypes` (easiest path) or
b) it looks like we always return a html element via `createElementFromTemplateResult`, so maybe the `undefined` check is not even needed? In this case, we could return the result from `createElementFromTemplateResult` directly.

Reply:
I would go for a).

If someone is doing something like `<div>My content</div>` in the `getOption` method, it is a htmlelement, but we should not allow this. It would break other things (they would have broken before, but we had no check). Now I react in the code itself by just using the "original" syn-option without the "getOption" changes, if it is no syn-option element: 
![image](https://github.com/user-attachments/assets/5616b763-66b2-4e09-86fe-c857745ebeb1)

 And we also state that only syn-options are allowed in the default slot: 
![image](https://github.com/user-attachments/assets/5b23afd6-1ce6-439e-bade-7a10afdefbf0)